### PR TITLE
V3 add user comments to timeline

### PR DESCRIPTION
**Pull request by spclu6**
#### What problem is this solving?

Add the user comments to the return request timeline when the request is created

#### How to test it?

[Workspace](https://returnappusercomments--powerplanet.myvtex.com/admin)

#### Screenshots or example usage:

Screenshot 2022-06-21 at 10 24 38

### CODE REVIEW

**Review by filafb — CHANGES_REQUESTED** [review]
Nice work! Only I thing. We can remove `userComment` from the master data schema [here](https://github.com/vtex-apps/return-app/blob/124400a40a1b3b31e4264b365e87d3c796cc5513/masterdata/returnRequest/schema.json#L151).

After doing it, please relink the project and run `vtex setup typings`to make sure all types are in place.

**Comment by filafb** [comment]
@cesarocampov what do you think about this UI for user comment? Instead of having it in a different component, we placed it inside the status timeline, creating a single flow for comments, either from store user and admin user.

**Comment by cesarocampov** [comment]
> @cesarocampov what do you think about this UI for user comment? Instead of having it in a different component, we placed it inside the status timeline, creating a single flow for comments, either from store user and admin user.

@filafb @spclu6 I like this approach. 
My only comment would be if we could enable a tag other than the email to identify Customer and Operator responses. 
take a look at the image below:
- the first comment was opened by me on the request. it used my email, this is correct. 
- the second comment was made in the admin of the app as an operator but added the Name of the customer, this is incorrect. 

Screenshot 2022-06-22 at 15 32 55
can we add a string before the comment stating
Customer - email
Operator - email?
what do you think?

**Comment by filafb** [comment]
> the second comment was made in the admin of the app as an operator but added the Name of the customer, this is incorrect.
We get this value from session. Is it possible you were logged in the same browser as store user when you submitted the second comment?

>  can we add a string before the comment stating Customer - email Operator - email? what do you think? 
Let's do it. @spclu6 I will reach out to you on slack to discuss this.

**Review by filafb — CHANGES_REQUESTED** [review]

**Comment by spclu6** [comment]
I added the user role into the return request comments so we can see which user type has made the comment.

Screenshot 2022-06-23 at 12 24 03

File: masterdata/returnRequest/schema.json [hunk 1]
```diff
@@ -211,6 +210,7 @@
                   "format": "date-time",
                   "required": true
                 },
+                "role": { "type": "string", "required": true },
```
Comment (filafb) [inline masterdata/returnRequest/schema.json:213]: You can make it as ENUM too. It adds one more layer of check for us.

**Review by filafb — CHANGES_REQUESTED** [review]
Really nice! I love it. @cesarocampov What do you think?

Just asking to make the type on the schema leval a `enum` too.

File: masterdata/returnRequest/schema.json (hunk 1 above)
Comment (cesarocampov) [reply masterdata/returnRequest/schema.json:213]: @spclu6 nice!

**Review by filafb — APPROVED** [review]

**Pull request merged**